test/e2e/case10_policyset_propagation_test.go

```diff
@@ -899,4 +901,157 @@ var _ = Describe("Test policyset propagation", func() {
+			plc = utils.GetWithTimeout(
+				clientHubDynamic, gvrPolicy, testNamespace+"."+case10PolicyName+"2", "managed1", true,
+				defaultTimeoutSeconds,
+			)
+			Expect(plc).ToNot(BeNil())
+			opt = metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + testNamespace + "." + case10PolicyName + "2",
+			}
+			utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 1, true, defaultTimeoutSeconds)
+		})
+		It("should propagate to cluster ns managed2", func() {
+			By("Patching " + case10PolicySetName + "-plm with decision of cluster managed2")
+			plm := utils.GetWithTimeout(
+				clientHubDynamic, gvrPlacementDecision, case10PolicySetName+"-plm-decision", testNamespace, true,
+				defaultTimeoutSeconds,
+			)
+			plm.Object["status"] = utils.GeneratePldStatus(plm.GetName(), plm.GetNamespace(), "managed2")
+			_, err := clientHubDynamic.Resource(gvrPlacementDecision).Namespace(testNamespace).UpdateStatus(
+				context.TODO(), plm, metav1.UpdateOptions{},
+			)
+			Expect(err).To(BeNil())
+			plc := utils.GetWithTimeout(
+				clientHubDynamic, gvrPolicy, testNamespace+"."+case10PolicyName+"1", "managed1", false,
+				defaultTimeoutSeconds,
+			)
+			Expect(plc).To(BeNil())
+			plc = utils.GetWithTimeout(
+				clientHubDynamic, gvrPolicy, testNamespace+"."+case10PolicyName+"1", "managed2", true,
+				defaultTimeoutSeconds,
+			)
+			Expect(plc).NotTo(BeNil())
+			plc = utils.GetWithTimeout(
+				clientHubDynamic, gvrPolicy, testNamespace+"."+case10PolicyName+"2", "managed1", false,
+				defaultTimeoutSeconds,
+			)
+			Expect(plc).To(BeNil())
+			plc = utils.GetWithTimeout(
+				clientHubDynamic, gvrPolicy, testNamespace+"."+case10PolicyName+"2", "managed2", true,
+				defaultTimeoutSeconds,
+			)
+			Expect(plc).NotTo(BeNil())
+		})
+		It("should still work when pb contains non-existing policyset", func() {
+			By("Patching " + case10PolicySetName + "-pb with non-existing policyset")
+			unstructuredPb := utils.GetWithTimeout(
+				clientHubDynamic, gvrPlacementBinding, case10PolicySetName+"-pb", testNamespace, true,
+				defaultTimeoutSeconds,
+			)
+			var pb policiesv1.PlacementBinding
+			err := runtime.DefaultUnstructuredConverter.
+				FromUnstructured(unstructuredPb.UnstructuredContent(), &pb)
+			Expect(err).To(BeNil())
+			nonExistingSubject := []policiesv1.Subject{
+				{
+					APIGroup: "policy.open-cluster-management.io",
+					Kind:     "PolicySet",
+					Name:     case10PolicySetName,
+				},
+			}
+			unstructuredPb.Object["subjects"] = append(nonExistingSubject, pb.Subjects[0], pb.Subjects[1])
+			_, err = clientHubDynamic.Resource(gvrPlacementBinding).Namespace(testNamespace).Update(
+				context.TODO(), unstructuredPb, metav1.UpdateOptions{},
+			)
+			Expect(err).To(BeNil())
+		})
+		It("should still propagate to cluster ns managed1", func() {
+			By("Patching " + case10PolicySetName + "-plm with decision of cluster managed1")
```

Comment:
Would this have been already propagated from the `should propagate to cluster ns managed1` test? I don't see any code between that test and this one that undoes the propagation.

Reply:
`should propagate to cluster ns managed2` on line 952 removed the policy from `managed1` and propagated to `managed2`